src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1897,6 +1906,7 @@ private unsafe void WriteEventVarargs(int eventId, Guid* childActivityID, object
                 Debug.Assert(m_eventData != null);  // You must have initialized this if you enabled the source.
                 try
                 {
+                    m_activeWritesCount++;
```

Comment:
Same comment as above, any time you increase m_activeWritesCount you have to immediately check m_eventSourceEnabled and bail if it's false

Reply:
There are a couple more places we don't check m_eventSourceEnabled, I'm not going to comment on every one 😄